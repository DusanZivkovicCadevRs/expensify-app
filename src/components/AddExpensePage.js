import React from 'react';

const AddExpensePage = () => (
    <div>
        Add this expense
    </div>
);

export default AddExpensePage;