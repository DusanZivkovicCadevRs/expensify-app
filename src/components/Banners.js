import React from 'react';

export default class Banner extends React.Component {

    state = {
        joke: ''
    }


    getJoke = () => {
        fetch('https://picsum.photos/200/300/?random')
            .then(res => res.json())
            .then(res => { console.log(JSON.stringify(res)); this.setState(() => ({ joke: res })) })
    }

    getBannerImage = () => {
        return (
            <div>
                <img src='https://picsum.photos/200/300/?random' />
            </div>
        );
    }

    render() {
        return (
            <div>
                <h3>Superior Banner</h3>
                <img src='https://picsum.photos/300/200/?random' />
            </div>
        );
    }
}