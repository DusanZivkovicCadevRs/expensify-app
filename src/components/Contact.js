import React from 'react';

const Contact = () => (
    <div>
        <p>Im Dule and my contact is mail</p>
    </div>
);

export default Contact;
