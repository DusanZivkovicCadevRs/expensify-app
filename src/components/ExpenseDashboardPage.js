import React from 'react';
import ExpenseList from './ExpenseList';

const ExpenseDashboardpage = () => (
    <div>
        <ExpenseList />
    </div>
);

export default ExpenseDashboardpage;
