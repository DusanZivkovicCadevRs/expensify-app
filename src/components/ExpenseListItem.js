import React from 'react';
import { Link } from 'react-router-dom';

// Export a stateless functional component
// description, amount, createdAt

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`} >
            <h1>{description}</h1>
        </Link>
        <p>{amount} - {createdAt}</p>

    </div>
);

export default ExpenseListItem;
