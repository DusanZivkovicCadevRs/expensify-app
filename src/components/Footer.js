import React from 'react';

export default class Footer extends React.Component {

    state = {
        joke: ''
    }


    getJoke = () =>{
        fetch('https://api.chucknorris.io/jokes/random')
            .then(res => res.json())
            .then(res => { console.log(JSON.stringify(res)); this.setState(() => ({ joke: res.value })) })
    }

    render() {
        return (
            <div>
                <h3>Really cool Footer</h3>
                <button onClick={this.getJoke}>Get Joke</button>
                <p>{this.state.joke}</p>
            </div>
        );
    }
}