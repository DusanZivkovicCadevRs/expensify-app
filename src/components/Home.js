import React from 'react';

const Home = () => (
    <div>
        <h1>Home page</h1>
        <p>wellcome here</p>
    </div>
);

export default Home;
