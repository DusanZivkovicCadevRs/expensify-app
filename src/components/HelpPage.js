import React from 'react';
import Footer from './Footer';
import Banners from './Banners';

const HelpPage = () => (
    <div>
        <Banners />
        Help me page
        <Footer />
    </div>
);

export default HelpPage;
