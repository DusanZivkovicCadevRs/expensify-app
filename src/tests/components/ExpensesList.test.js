import React from 'react';
import {shallow} from 'enzyme';
import expensesFixture from '../fixtures/expenses';
import {ExpenseList} from '../../components/ExpenseList';

test('should render ExpensesList with expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expensesFixture}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesList without expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
});
