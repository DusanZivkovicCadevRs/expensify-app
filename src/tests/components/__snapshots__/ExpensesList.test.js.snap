// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesList with expenses 1`] = `
<div>
  <h1>
    ExpenseList
  </h1>
  <ExpenseListItem
    amount={169}
    createdAt={0}
    description="Bill"
    endDate={259200000}
    id="0"
    key="0"
    note=""
  />
  <ExpenseListItem
    amount={16900}
    createdAt={-345600000}
    description="Rent"
    endDate={432000000}
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={55555}
    createdAt={345600000}
    description="Credit card"
    endDate={432000000}
    id="2"
    key="2"
    note=""
  />
</div>
`;

exports[`should render ExpensesList without expenses 1`] = `
<div>
  <h1>
    ExpenseList
  </h1>
  <p>
    No expenses
  </p>
</div>
`;
